bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/OpenBuilder.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.BKException;
+
+/**
+ * Builder-style interface to open exiting ledgers
+ */
+public interface OpenBuilder extends OpBuilder<ReadHandler> {
+
+    OpenBuilder withLedgerId(long ledgerId);
+
+    OpenBuilder withRecovery(boolean recovery);
+
+    OpenBuilder withPassword(byte[] password);
+
+    OpenBuilder withDigestType(org.apache.bookkeeper.client.BookKeeper.DigestType digestType);
+
+    CompletableFuture<ReadHandler> execute();
+
+    ReadHandler open() throws BKException, InterruptedException;
```

Comment:
still keeping some sync methods?

Reply:
I kept those methods in order to have simpler try-with-resources blocks
try (ReadHandler reader = bk.openLedgerOp().withLedgerId(id).open()) {
        reader.read(xxx);
}

instead of:
try (ReadHandler reader = FutureUtils.result(bk
            .openLedgerOp()
            .withLedgerId(id)
            .execute())) {
        reader.read(xxx);
}

I can drop them either if you prefer

Reply:
No strong opinion, I just thought you had removed them given the earlier message.

Reply:
I think in this specific cases (open and create) it is better to keep them, these are the first APIs seen by newbies

try (WriteHandler w = bk.createLedgerOp().create() {
       w.append(data);
       w.append(data);
       w.append(data);
       w.append(data).get();
}

/cc @sijie

Reply:
sounds good to me.